internal/controller/controllers/agentserviceconfig_controller.go

```diff
@@ -295,15 +304,62 @@ func (r *AgentServiceConfigReconciler) ensureAssistedServiceDeployment(ctx conte
 	return nil
 }
 
+func (r *AgentServiceConfigReconciler) ensureIngressCertCM(ctx context.Context, instance *aiv1beta1.AgentServiceConfig) error {
+	sourceCM := &corev1.ConfigMap{}
+
+	if err := r.Get(ctx, types.NamespacedName{Name: defaultIngressCertCMName, Namespace: defaultIngressCertCMNamespace}, sourceCM); err != nil {
+		r.Log.Error(err, "Failed to get default ingress cert config map")
+		conditionsv1.SetStatusCondition(&instance.Status.Conditions, conditionsv1.Condition{
+			Type:    aiv1beta1.ConditionReconcileCompleted,
+			Status:  corev1.ConditionFalse,
+			Reason:  aiv1beta1.ReasonDeploymentFailure,
+			Message: "Failed to get default ingress cert config map: " + err.Error(),
+		})
+		return err
+	}
+
+	cm, mutateFn := r.newIngressCertCM(instance, sourceCM)
+
+	if result, err := controllerutil.CreateOrUpdate(ctx, r.Client, cm, mutateFn); err != nil {
+		conditionsv1.SetStatusCondition(&instance.Status.Conditions, conditionsv1.Condition{
+			Type:    aiv1beta1.ConditionReconcileCompleted,
+			Status:  corev1.ConditionFalse,
+			Reason:  aiv1beta1.ReasonDeploymentFailure,
+			Message: "Failed to ensure ingress cert config map: " + err.Error(),
+		})
+		return err
+	} else if result != controllerutil.OperationResultNone {
+		r.Log.Info("Ingress config map created")
+	}
+	return nil
+}
+
+func checkIngressCMName(obj metav1.Object) bool {
+	return obj.GetNamespace() == defaultIngressCertCMNamespace && obj.GetName() == defaultIngressCertCMName
+}
+
 // SetupWithManager sets up the controller with the Manager.
 func (r *AgentServiceConfigReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	ingressCMPredicates := builder.WithPredicates(predicate.Funcs{
+		CreateFunc:  func(e event.CreateEvent) bool { return checkIngressCMName(e.Object) },
+		UpdateFunc:  func(e event.UpdateEvent) bool { return checkIngressCMName(e.ObjectNew) },
+		DeleteFunc:  func(e event.DeleteEvent) bool { return checkIngressCMName(e.Object) },
+		GenericFunc: func(e event.GenericEvent) bool { return checkIngressCMName(e.Object) },
+	})
+	ingressCMHandler := handler.EnqueueRequestsFromMapFunc(
+		func(_ client.Object) []reconcile.Request {
+			return []reconcile.Request{{NamespacedName: types.NamespacedName{Name: agentServiceConfigName}}}
```

Comment:
Do we not need to provide the namespace here?

Reply:
I was curious about that, but @djzager said that the agent service config was not namespaced.

Also, I'm not sure how we would get access to the namespace (assuming it's not fixed).

Reply:
Ah right, makes sense.